pcsound/pcsound_sdl.c

```diff
@@ -78,19 +78,14 @@ static void PCSound_Mix_Callback(void *udata, Uint8 *stream, int len)
 
         while (current_remaining == 0) 
         {
-            oldfreq = current_freq;
-
             // Get the next frequency to play
 
-            callback(&current_remaining, &current_freq);
+            callback(&current_remaining, &frequency);
 
-            if (current_freq != 0)
+            if (current_freq != frequency)
             {
-                // Adjust phase to match to the new frequency.
-                // This gives us a smooth transition between different tones,
-                // with no impulse changes.
-
-                phase_offset = (phase_offset * oldfreq) / current_freq;
+                current_freq = frequency;
+                phase_offset = 0;
```

Comment:
Why do you reset the phase offset after changing frequency? This is supposed to make the transition between tones less harsh.